### DISCUSSION TITLE
HP_ERM encapsulation support

### DISCUSSION THREAD

**Discussion by ChicagoJay** [sipcapture/homer#457] [Q&A] [answered]
I think I'm missing a step.  I have my switch mirroring data from the voice gateway (a CUBE router) to the homer server (a VMWare vm, running Debian 10), which I set up using the "Quick install".  

Everything appears to be up and running.  I loaded tcpdump, and I see the SIP traffic coming in (although it's encapsulated, because that's the only way to get Aruba to mirror to an IP address).  The packets all come from the switch's IP port 9060 to homer's IP on port 7932.  Unfortunately, there is no way to get the Aruba switch to send packets to any other port.  

How can I get heplify to listen to port 7932 and push that traffic into something homer can see?  Right now, homer sees no traffic.

Thanks!

**Comment by lmangani** [comment, marked as answer]
HOMER only understands HEP and no other protocols. All the work is done by the agents in charge of the various protocols.
What aruba encapsulation are you referring to here? What is coming to port 7932? Try to be as specific as possible so we can offer back potential routes. I think the CUBE should be able to do erspan in which case just use the -erspan flag in heplify to have it captured.

**Reply by ChicagoJay** [reply]
So our CUBE router (ISR 4321) won't do ERSPAN on the built-in port - we'd need to install a switch module.  So I have the Aruba, which is between the ISR and the SIP provider, doing the mirroring.  I think the protocol is called HP_ERM protocol, there is an in-depth explanation of how it works, and how to analyze it in Wireshark, [here](https://community.arubanetworks.com/blogs/esupport1/2019/10/30/how-to-analyze-and-send-remote-mirrored-traffic-directly-to-a-pcanalyzer-running-wireshark).  [Here](https://wiki.wireshark.org/HPERM) is Wireshark's writeup on it.

Thanks again!

**Reply by ChicagoJay** [reply]
I just thought of something:  Can I put ERSPAN on the SIP ports, on the CUBE router, and have those go to heplify?
-- edit --
Nope - one may not use the interface being monitored to send mirror traffic.  So, since I have 2 SIP ports coming in, I can't use one of the ports to mirror data.  Bummer.

**Reply by kYroL01** [reply]
I analyzed the pcap - the `HP_ERM` is a 12 bytes layer and the real packet is encapsulated into the "fake" packet after this Aruba layer - the port `7932` it's the official one. basically is the same way to encapsulate a packet like `GRE/ERSPAN` does, but I think is not so frequent to see, that's why we don't have a support for that yet.

**Reply by ChicagoJay** [reply]
@kYroL01 thanks for the analysis!  Is that something that may be implemented soon?  Is there another way for me to make this work, like a service that can convert the packets from `HP_ERM` to `ERSPAN` and put them on the correct port `9060`?

I'm a `SIP` n00b, so sorry if that seems like a preposterous ask....

**Reply by lmangani** [reply]
@ChicagoJay would you be interested in sponsoring this feature?

**Reply by ChicagoJay** [reply]
> @ChicagoJay would you be interested in sponsoring this feature?

A fair question, @lmangani!  What would be involved?  Very happy to test and give feedback, screenshots, logs, etc...

**Reply by lmangani** [reply]
@ChicagoJay anyone can help us with donations using the [made by humans](https://github.com/sipcapture/homer#made-by-humans) link in most of our projects. I would suggest opening a [feature request](https://github.com/sipcapture/homer/issues) with all the required details + a donation of your choice for a best effort initiative and no guarantees - mind this will require hours of actual engineering and development work as you consider this step. If you would prefer this done professionally you can contact [qxip](mailto:sales@qxip.net) for a quotation instead. Thanks!

**Reply by ChicagoJay** [reply]
@lmangani - I work for a school, on a limited budget, so I just chipped in US$5 out of my own pocket.  I know it's not much, but at least it'll buy someone a coffee....

**Reply by lmangani** [reply]
@ChicagoJay we appreciate the offer, if you can prove you work for a non profit contact qxip and we'll be glad to help

**Reply by lmangani** [reply]
To be continued in https://github.com/sipcapture/homer/issues/458

**Reply by ChicagoJay** [reply]
> @ChicagoJay we appreciate the offer, if you can prove you work for a non profit contact qxip and we'll be glad to help

@lmangani Check your email...
